Sources/ImageFetcher/ImageFetcher.swift

```diff
@@ -171,30 +184,25 @@ public extension ImageFetcher {
     /// Cancels an in-flight image load
     /// - Parameter imageConfiguration: The configuation of the image to be downloaded.
     func cancel(_ imageConfiguration: ImageConfiguration) {
-        guard let task = self[imageConfiguration] else {
+        guard let task = tasks[imageConfiguration.key], task.isPending else {
```

Comment:
I opted against the subscript since we  have to use `tasks` when removing the fetcher task below